src/parser/shaman/enhancement/modules/talents/LightningShield.js

```diff
@@ -1,45 +1,44 @@
 import React from 'react';
 import SPELLS from 'common/SPELLS/index';
-import SpellIcon from 'common/SpellIcon';
 import { formatNumber, formatPercentage } from 'common/format';
 
 import Analyzer from 'parser/core/Analyzer';
 
-import StatisticBox from 'interface/others/StatisticBox';
-import SpellLink from 'parser/shaman/elemental/modules/talents/Icefury';
+import TalentStatisticBox from 'interface/others/TalentStatisticBox';
+import SpellLink from 'common/SpellLink';
 
 class LightningShield extends Analyzer {
-
-  damageGained=0;
-  overchargeCount=0;
+  damageGained = 0;
 
   constructor(...args) {
     super(...args);
     this.active = this.selectedCombatant.hasTalent(SPELLS.LIGHTNING_SHIELD_TALENT.id);
   }
 
+  isEventLightningShield(event) {
+    switch (event.ability.guid) {
+      case SPELLS.LIGHTNING_SHIELD_OVERCHARGE.id:
+      case SPELLS.LIGHTNING_SHIELD.id:
+        return true;
+      default:
+        return false;
+    }
+  }
+
   on_byPlayer_damage(event) {
-    if (event.ability.guid!==SPELLS.LIGHTNING_SHIELD_TALENT.id) {
+    if (this.isEventLightningShield(event) === false) {
```

Comment:
is there a reason to add more lines of code to achieve the same functionality, and make it non-consistent with other modules?

Reply:
I was kind of getting used to how the event handlers worked. This has been changed. in the current version. Newest verion which i'm working on works with typescript. (Credits to Hawk)